src/requests.rs

```diff
@@ -141,7 +142,5 @@ impl RequestDelay {
 }
 
 pub(crate) fn response_info(resp: &mut reqwest::Response) -> (StatusCode, Url, HeaderMap) {
-    let mut headers = HeaderMap::new();
-    std::mem::swap(&mut headers, resp.headers_mut());
-    (resp.status(), resp.url().clone(), headers)
+    (resp.status(), resp.url().clone(), resp.headers_mut().clone())
```

Comment:
Was this swap just for performance reasons? It needs to return `headers_mut()` response's directly since it will be mutated and isn't just a view "stuck in time".